qiskit_ibm_provider/ibm_backend_service.py

```diff
@@ -149,11 +155,28 @@ def backends(
                 `project` are specified.
         """
         backends: List[IBMBackend] = []
+        ret = OrderedDict()
         if instance:
             hgp = self._provider._get_hgp(instance=instance)
-            backends = list(hgp.backends.values())
+            for backend_name in hgp.backends.keys():
+                if not self._backends[backend_name]:
+                    self._backends[backend_name] = self._fetch_backend_config(
+                        backend_name, instance
+                    )
```

Comment:
I think this is going to break the "honor instance specified" feature you added recently. For example, if I do

```
b1 = provider.backends(instance="h1/g1/p1")
b2 = provider.backends(instance="h2/g2/p2")
```
b2 is going to have IBMBackend instances initialized with h1/g1/p1 if a backend is in both.